### PR TITLE
fix: unset PODIO_JSON_OUTPUT after loading EDM4HEP, EDM4EIC

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This unsets PODIO_JSON_OUTPUT for the build, shaving 13% off the build time. See https://github.com/AIDASoft/podio/issues/475 for details. Addresses some of https://github.com/eic/EICrecon/issues/628 but not all (although, at what point do we close that issue? when it compiles in zero time?)

Before:
```
real    6m38.902s
user    47m34.090s
sys     2m52.019s
```

After:
```
real    5m53.079s
user    40m30.379s
sys     2m34.458s
```

### What kind of change does this PR introduce?
- [x] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,6 +98,15 @@ find_package(podio REQUIRED)
...
 
+# Remove PODIO_JSON_OUTPUT (ref: https://github.com/AIDASoft/podio/issues/475)
+get_target_property(EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EDM4HEP::edm4hep INTERFACE_COMPILE_DEFINITIONS)
+list(FILTER EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
+set_property(TARGET EDM4HEP::edm4hep PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4HEP_INTERFACE_COMPILE_DEFINITIONS})
+
+get_target_property(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EDM4EIC::edm4eic INTERFACE_COMPILE_DEFINITIONS)
+list(FILTER EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
+set_property(TARGET EDM4EIC::edm4eic PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4EIC_INTERFACE_COMPILE_DEFINITIONS})
+
```
Comment (wdconinc) [inline CMakeLists.txt:109]: Make this more robust.
```suggestion
if(TARGET EDM4EIC::edm4eic)
  get_target_property(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EDM4EIC::edm4eic INTERFACE_COMPILE_DEFINITIONS)
  if(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS)
    list(FILTER EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
    set_property(TARGET EDM4EIC::edm4eic PROPERTY INTERFACE_COMPILE_DEFINITIONS 
  ${EDM4EIC_INTERFACE_COMPILE_DEFINITIONS})
  endif()
endif()
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -98,6 +98,15 @@ find_package(podio REQUIRED)
 find_package(EDM4HEP REQUIRED)
 find_package(EDM4EIC 2.1 REQUIRED)
 
+# Remove PODIO_JSON_OUTPUT (ref: https://github.com/AIDASoft/podio/issues/475)
+get_target_property(EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EDM4HEP::edm4hep INTERFACE_COMPILE_DEFINITIONS)
+list(FILTER EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
+set_property(TARGET EDM4HEP::edm4hep PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4HEP_INTERFACE_COMPILE_DEFINITIONS})
+
```
Comment (wdconinc) [inline CMakeLists.txt:105]:
```suggestion
# Remove PODIO_JSON_OUTPUT (ref: https://github.com/AIDASoft/podio/issues/475)
if(TARGET EDM4HEP::edm4hep)
  get_target_property(EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EDM4HEP::edm4hep INTERFACE_COMPILE_DEFINITIONS)
  if(EDM4HEP_INTERFACE_COMPILE_DEFINITIONS)
    list(FILTER EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
    set_property(TARGET EDM4HEP::edm4hep PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4HEP_INTERFACE_COMPILE_DEFINITIONS})
  endif()
endif()
```

**Comment by wdconinc** [comment]
Clang is not happy with this: https://github.com/eic/EICrecon/actions/runs/6088605533/job/16519623963?pr=916#step:7:240, so needs better handling.

File: CMakeLists.txt [hunk 3]
```diff
@@ -98,6 +98,15 @@ find_package(podio REQUIRED)
...
+get_target_property(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EDM4EIC::edm4eic INTERFACE_COMPILE_DEFINITIONS)
+list(FILTER EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
+set_property(TARGET EDM4EIC::edm4eic PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4EIC_INTERFACE_COMPILE_DEFINITIONS})
```
Comment (wdconinc) [inline CMakeLists.txt:111]:
```suggestion
if(TARGET EDM4EIC::edm4eic)
  get_target_property(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EDM4EIC::edm4eic INTERFACE_COMPILE_DEFINITIONS)
  if(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS)
    list(FILTER EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
    set_property(TARGET EDM4EIC::edm4eic PROPERTY INTERFACE_COMPILE_DEFINITIONS 
  ${EDM4EIC_INTERFACE_COMPILE_DEFINITIONS})
  endif()
endif()
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -98,6 +98,23 @@ find_package(podio REQUIRED)
...
+    list(FILTER EDM4HEP_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
+    set_property(TARGET EDM4HEP::edm4hep PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4HEP_INTERFACE_COMPILE_DEFINITIONS})
+  endif()
+endif()
+if(TARGET EDM4EIC::edm4eic)
+  get_target_property(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EDM4EIC::edm4eic INTERFACE_COMPILE_DEFINITIONS)
+  if(EDM4EIC_INTERFACE_COMPILE_DEFINITIONS)
+    list(FILTER EDM4EIC_INTERFACE_COMPILE_DEFINITIONS EXCLUDE REGEX [[^PODIO_JSON_OUTPUT$]])
+    set_property(TARGET EDM4EIC::edm4eic PROPERTY INTERFACE_COMPILE_DEFINITIONS
+  ${EDM4EIC_INTERFACE_COMPILE_DEFINITIONS})
```
Comment (wdconinc) [inline CMakeLists.txt:114]:
```suggestion
    set_property(TARGET EDM4EIC::edm4eic PROPERTY INTERFACE_COMPILE_DEFINITIONS ${EDM4EIC_INTERFACE_COMPILE_DEFINITIONS})
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -98,6 +98,22 @@ find_package(podio REQUIRED)
 find_package(EDM4HEP REQUIRED)
 find_package(EDM4EIC 2.1 REQUIRED)
 
+# Remove PODIO_JSON_OUTPUT (ref: https://github.com/AIDASoft/podio/issues/475)
+if(TARGET EDM4HEP::edm4hep)
```
Comment (veprbl) [inline CMakeLists.txt:102]: Should not be conditional. If target is renamed, we'd like for configuration to fail loudly and allow us to adapt the workaround.

Comment (wdconinc) [reply CMakeLists.txt]: Addressed in https://github.com/eic/EICrecon/pull/916/commits/7262ca69936a65c1c18959e7653d5a790c8b6f5c.

**Review by veprbl — APPROVED** [review]
LGTM

**Pull request merged**